cmd/registry/patch/apply.go

```diff
@@ -44,29 +44,31 @@ func Apply(ctx context.Context, client connection.RegistryClient, in io.Reader,
 		return patches.run(ctx, jobs)
 	}
 
-	err := filepath.WalkDir(path,
-		func(fileName string, entry fs.DirEntry, err error) error {
-			if err != nil {
+	for _, p := range paths {
+		err := filepath.WalkDir(p,
+			func(fileName string, entry fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				} else if entry.IsDir() && fileName != p && !recursive {
+					return filepath.SkipDir // Skip the directory and contents.
+				} else if entry.IsDir() {
+					return nil // Do nothing for the directory, but still walk its contents.
+				} else if !strings.HasSuffix(fileName, ".yaml") && !strings.HasSuffix(fileName, ".yml") {
+					return nil // Skip everything that's not a YAML file.
+				}
+				bytes, err := os.ReadFile(fileName)
+				if err != nil {
+					return err
+				}
+				err = patches.parse(client, bytes, fileName, project)
+				if err != nil {
+					err = fmt.Errorf("parsing %s: %w", fileName, err)
+				}
 				return err
-			} else if entry.IsDir() && fileName != path && !recursive {
-				return filepath.SkipDir // Skip the directory and contents.
-			} else if entry.IsDir() {
-				return nil // Do nothing for the directory, but still walk its contents.
-			} else if !strings.HasSuffix(fileName, ".yaml") && !strings.HasSuffix(fileName, ".yml") {
-				return nil // Skip everything that's not a YAML file.
-			}
-			bytes, err := os.ReadFile(fileName)
-			if err != nil {
-				return err
-			}
-			err = patches.parse(client, bytes, fileName, project)
-			if err != nil {
-				err = fmt.Errorf("parsing %s: %w", fileName, err)
-			}
+			})
+		if err != nil {
 			return err
-		})
-	if err != nil {
-		return err
+		}
 	}
 	return patches.run(ctx, jobs)
```

Comment:
Just double-checking: because we collect all the inputs before we call this, there's no sensitivity to the ordering of file names, right?

Reply:
What is your specific concern?

Reply:
I wouldn't want a child resource to fail to apply because its parent hasn't been applied yet. That's already addressed in `patches.run()` so since we're only callling `patches.run()` once for all the inputs, we should be good.

Reply:
OK, sg. We might not be checking the dependency in the tests.